Improve errors for invalid IDs in content collections (#12892)
withastro/astro: .changeset/dry-dragons-shout.md

---
'astro': patch
---

> Added more verbose errors for content collections with invalid IDs.
> 
> Previously, when an entry in a content collection had an ID that wasn't a string, the error would omit the ID entirely. With this change, the error will now explicitly say that the ID has to be a string.

Reviewer: ```suggestion
Adds a more descriptive error when a content collection entry has an invalid ID.
```

Just suggesting that I don't think a more elaborate description is necessary for a patch!
